onnx/backend/test/case/node/sequenceinsert.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np  # type: ignore
+
+import onnx
+from ..base import Base
+from . import expect
+
+
+def sequence_insert_reference_implementation(sequence, tensor, position=None):  # type: (np.ndarray, np.ndarray, np.ndarray) -> np.ndarray
+    sequence = list(sequence)
+    if position:
+        # In these cases, insert_position will be between [-len(sequence), len(sequence)]
+        # The position argument will be in the format np.array([pos_index])
+        insert_position = position[0]
+        sequence.insert(insert_position, tensor)
+    else:
+        # Default position of insertion is at the end of the sequence.
+        sequence.append(tensor)
+    return np.array(sequence)
+
+
+class SequenceInsert(Base):
+
+    @staticmethod
+    def export():  # type: () -> None
+        test_cases = {
+            'at_back': [np.array([10, 11, 12])],
+            'at_front': [np.array([-2, -1, 0]), np.array([0])]
+        }
+        sequence = np.array([np.array([1, 2, 3, 4]), np.array([5, 6, 7]), np.array([8, 9])])
```

Comment:
It would be great if we could just write sequence = [ np.array(...), np.array(...), ...] here. It also clarifies that it is a sequence of tensors.

Reply:
Addressed. Thanks!

Reply:
I am a bit surprised that this did not require other changes (as discussed elsewhere above about whether we should use arrays or lists for representing sequences)

Reply:
Likewise. This ended up being a point in test implementation, not test infrastructure as I had previously thought. I'll test with ORT and see if it works. Thanks!

Reply:
The changes I was talking about earlier were in the helper functions (i.e. from_array_to_sequence, etc.) Will submit an update.

Reply:
I think changes might be required in _extract_value_info (as I have commented there) also to make this work.